ql/src/java/org/apache/hadoop/hive/ql/exec/FunctionRegistry.java

```diff
@@ -289,6 +297,16 @@
     system.registerGenericUDF("trunc", GenericUDFTrunc.class);
     system.registerGenericUDF("date_format", GenericUDFDateFormat.class);
 
+    // Special date formatting functions
```

Comment:
These are equivalent to FLOOR(date to TIMEUNIT). E.g. year_granularity(date) is equivalent to FLOOR(date to YEAR).

In turn, extract is equivalent to UDFs that are still present in Hive. E.g. year(date) is equivalent to EXTRACT(YEAR from date).

Both collections of UDFs will stay there; we just need to link them to the right syntax in the parser.

As part of the work to remove the dependency on Druid in ql, I will bring the code of the underlying logic for these UDFs from Druid to Hive (I have checked and they code is quite self-contained).